CNV#27695: Update RN known issues and bug fixes for 4.15 (#71598)
openshift/openshift-docs: virt/release_notes/virt-4-15-release-notes.adoc

[id="virt-4-15-bug-fixes"]
== Bug fixes

> * Previously, EFI pipelines failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])

Reviewer: @sjhala-ccs @dominikholler this is old news, but I don't know what are `EFI pipelines`. Can a URL be added?

Reviewer 2: ```suggestion
* Previously, the pipeline windows-efi-installer failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])
```

> * Previously, if you simultaneously cloned more than 1000 VMs using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])

Reviewer: ```suggestion
* Previously, if you simultaneously cloned more than ~1000 VMs using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
```

@aglitke I think that it's not about 1000 precisely

> * Previously, you could not SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. With this update, you can use a `NodePort` service for VM SSH access. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])

Reviewer: ```suggestion
* Previously, you could not SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. With this update, you can use a `NodePort` service for VM SSH access. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])
```
@phoracek I think this should be revised to refer to GUI and ease of use.

Author: @phoracek Do you have any suggestions to update the wording for the above issue?

Reviewer 3: I added a suggestion for an alternative text.

[id="virt-4-15-bug-fixes"]
== Bug fixes
* Previously, the `windows-efi-installer` pipeline failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])

> * Previously, if you simultaneously cloned approximately 1000 VMs using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])

Reviewer 4: Suggestion to spell out the first instance of VM in this section:
```suggestion
* Previously, if you simultaneously cloned approximately 1000 virtual machines (VMs) using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
```

> * Previously, you could not SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. With this update, you can use a `NodePort` service for VM SSH access. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])

Reviewer 3: ```suggestion
* Previously, you could not SSH into a VM over a NodePort Service and its associated FQDN displayed in the UI when using OVNKubernetes. With this update, you can configure Console so it shows a valid accessible endpoint for SSH NodePort Services. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])
```

Author: Updated per your suggestion

[discrete]
[id="virt-4-15-ki-monitoring"]
==== Monitoring
//fix targeted for 4.16
* The Pod Disruption Budget (PDB) prevents pod disruptions for migratable virtual machine images. If the PDB detects pod disruption, then `openshift-monitoring` sends a `PodDisruptionBudgetAtLimit` alert every 60 minutes for virtual machine images that use the `LiveMigrate` eviction strategy. (link:https://issues.redhat.com/browse/CNV-33834[*CNV-33834*])
** As a workaround, xref:../../monitoring/managing-alerts.adoc#silencing-alerts_managing-alerts[silence alerts].

//[discrete]
//[id="virt-4-15-ki-networking"]
//==== Networking

> //CNV-10590 - This issue can be removed for 4.15 when CNV-13680 RN is added.

Reviewer: @sjhala-ccs why is this commented out and not removed? (really asking, no idea)

Author: Was planning to remove it after I added in the release note for CNV-13680. Sorry about the confusion.

//* If your {product-title} cluster uses OVN-Kubernetes as the default Container Network Interface (CNI) provider, you cannot attach a Linux bridge or bonding device to a host's default interface because of a change in the host network topology of OVN-Kubernetes. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1885605[*BZ#1885605*])
//** As a workaround, you can use a secondary network interface connected to your host, or switch to the OpenShift SDN default CNI provider.

[discrete]
[id="virt-4-15-ki-nodes"]
==== Nodes
//4.15 Check with Stu
//Need link to engineering bug instead of docs bug

> * Uninstalling {VirtProductName} does not remove the `feature.node.kubevirt.io` node labels created by {VirtProductName}. You must remove the labels manually.

Author: @stu-gott Is there an engineering bug that I can link to for the above issue?

Reviewer 5: /cc @tiraboschi

I think HCO is the component that creates this label. Can you point @sjhala-ccs in the right direction?

Reviewer 6: I think that it's added by node-labeller executed by virt-handler: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/node-labeller/node_labeller.go#L249

Author: @tiraboschi So the labels still need to be removed manually and this bug applies to 4.15? Can you provide a link to the actual bug? Thanks!

Reviewer 5: @sjhala-ccs Neither Virtualization nor Install, Upgrade and Operators components appear to have a relevant bug for this.

Reviewer 5: I'll need 24 hours to validate if this issue exists.

Reviewer 5: OK, it looks like those labels are not cleaned up. The RN is correct.

Reviewer 5: I've added a dev tracker to clean this up.

https://issues.redhat.com/browse/CNV-38543

Author: Thanks. I will add that link to the RN.

[discrete]
[id="virt-4-15-ki-storage"]
==== Storage
//4.15 Check with Alex Kalenyuk. Issue is closed as "Not a bug"
* If you use Portworx as your storage solution on AWS and create a VM disk image, the created image might be smaller than expected due to the filesystem overhead being accounted for twice. (link:https://issues.redhat.com/browse/CNV-32695[*CNV-32695*])
** As a workaround, you can manually expand the Persistent Volume Claim (PVC) to increase the available space after the initial provisioning process completes.

> //4.15 Check with Adam. Issue is closed as "Won't do" but is still a known issue

Author: @aglitke Should we still keep this issue for 4.15?

* In some instances, multiple virtual machines can mount the same PVC in read-write mode, which might result in data corruption. (link:https://issues.redhat.com/browse/CNV-13500[*CNV-13500*])
** As a workaround, avoid using a single PVC in read-write mode with multiple VMs.

> //4.15 Have asked Dominik Holler if we still need to keep this

Author: @dominikholler Can you please confirm if we need to keep this issue for 4.15?

* If you clone more than 100 VMs using the `csi-clone` cloning strategy, then the Ceph CSI might not purge the clones. Manually deleting the clones might also fail. (link:https://issues.redhat.com/browse/CNV-23501[*CNV-23501*])
** As a workaround, you can restart the `ceph-mgr` to purge the VM clones.

[discrete]
[id="virt-4-15-ki-virtualization"]
==== Virtualization
//New for 4.15

> * CPU hot unplug for Windows VMs is not supported and results in an error. 

Reviewer: This may be misleading - we don't yet support any kind of cpu hotplug. @stu-gott

Author: @stu-gott Thoughts on the above issue?

Reviewer 5: Concur that this note shouldn't be mentioned in 4.15. CPU Hotplug will GA in 4.16.

Hot unplug not working in Windows is a limitation of that OS. Unsure if that warrants a RN over something in docs as it's a permanent limitation outside of our control.

// Cannot link to issues with security level "Red Hat Employee" (link:https://issues.redhat.com/browse/CNV-37346[*CNV-37346*])

* When adding a virtual Trusted Platform Module (vTPM) device to a Windows VM, the BitLocker Drive Encryption system check passes even if the vTPM device is not persistent. This is because a vTPM device that is not persistent stores and recovers encryption keys using ephemeral storage for the lifetime of the `virt-launcher` pod. When the VM migrates or is shut down and restarts, the vTPM data is lost. (link:https://issues.redhat.com/browse/CNV-36448[*CNV-36448*])

//Looks like this is still unresolved for 4.15
* {VirtProductName} links a service account token in use by a pod to that specific pod. {VirtProductName} implements a service account volume by creating a disk image that contains a token. If you migrate a VM, then the service account volume becomes invalid. (link:https://issues.redhat.com/browse/CNV-33835[*CNV-33835*])
** As a workaround, use user accounts rather than service accounts because user account tokens are not bound to a specific pod.

> //4.15 Check with Stu if this needs to be moved to CNV docs

Author: @stu-gott Do we still need to keep the below known issue in the release notes?

Reviewer 5: The release note still applies. We are currently working toward virtiofs support in KubeVirt, which is the precursor to addressing this. Rough ETA for fixing this issue is forecast as 4.18+. If it's a problem to leave a RN in for that long we could consider moving this to somewhere in the documentation, but we do plan to fix it once we're able.

Author: Thanks for the update. We will leave the RN as is.

** As a workaround, you can restart the `ceph-mgr` to purge the VM clones.

[discrete]
[id="virt-4-15-ki-virtualization"]
==== Virtualization
//4.15 still unresolved
* When adding a virtual Trusted Platform Module (vTPM) device to a Windows VM, the BitLocker Drive Encryption system check passes even if the vTPM device is not persistent. This is because a vTPM device that is not persistent stores and recovers encryption keys using ephemeral storage for the lifetime of the `virt-launcher` pod. When the VM migrates or is shut down and restarts, the vTPM data is lost. (link:https://issues.redhat.com/browse/CNV-36448[*CNV-36448*])

//4.15 still unresolved 
* {VirtProductName} links a service account token in use by a pod to that specific pod. {VirtProductName} implements a service account volume by creating a disk image that contains a token. If you migrate a VM, then the service account volume becomes invalid. (link:https://issues.redhat.com/browse/CNV-33835[*CNV-33835*])
** As a workaround, use user accounts rather than service accounts because user account tokens are not bound to a specific pod.

//4.15 Keep this per Stu, ETA for fixing this issue is 4.18. 

> * With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (EMS) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled RHEL 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. 

Reviewer 4: Just curious, is there a reason that EMS is not in literals when the written out term is?
```suggestion
* With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (`EMS`) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled RHEL 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. 
```

Author: This is pre-existing content from 4.14.

> * With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (EMS) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled RHEL 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. 

Reviewer 4: I'd change this to the attribute, and possibly the "first mention" version as well:
```suggestion
* With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (EMS) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled {op-system-base-full} 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. 
```

Author: Pre-existing content. I am fine with updating it, but RHEL has been spelled out in the earlier sections.

+
Legacy OpenSSL clients that do not support EMS or TLS 1.3 now cannot connect to FIPS servers running on RHEL 9. Similarly, RHEL 9 clients in FIPS mode cannot connect to servers that only support TLS 1.2 without EMS. This in practice means that these clients cannot connect to servers on RHEL 6, RHEL 7 and non-RHEL legacy operating systems. This is because the legacy 1.0.x versions of OpenSSL do not support EMS or TLS 1.3. For more information, see link:https://access.redhat.com/solutions/7018256[TLS Extension "Extended Master Secret" enforced with Red Hat Enterprise Linux 9.2].

> ** As a workaround, upgrade legacy OpenSSL clients to a version that supports TLS 1.3 and configure {VirtProductName} to use TLS 1.3, with the `Modern` TLS security profile type, for FIPS mode.

Reviewer 4: Update is the preferred term per our [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#update):
```suggestion
** As a workaround, update legacy OpenSSL clients to a version that supports TLS 1.3 and configure {VirtProductName} to use TLS 1.3, with the `Modern` TLS security profile type, for FIPS mode.
```
